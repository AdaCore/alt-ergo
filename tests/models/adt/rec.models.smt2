(set-option :produce-models true)
(set-logic ALL)
(declare-datatype data (
  (cons1 (d1 data))
  (cons2)
))
(declare-const a data)
(assert ((_ is cons1) a))
(check-sat)
(get-model)
