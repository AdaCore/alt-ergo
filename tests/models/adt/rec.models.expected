
unknown
(
  (define-fun a () data (cons1 cons2))
)
